Assets/Scripts/Flock.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Flock : MonoBehaviour
+{
+    [SerializeField] private FlockAgent agentPrefab;
+    List<FlockAgent> agents = new List<FlockAgent>();
+    [SerializeField] private FlockBehaviour behaviour;
+
+    [Range(1, 500)]
+    [SerializeField] private int startingCount = 250;
+    const float AgentDensity = 0.08f;
+
+    [Range(1f, 100f)]
+    [SerializeField] private float driveFactor = 10f;
+    [Range(1f, 100f)]
+    [SerializeField] private float maxSpeed = 5f;
+    [Range(1f, 10f)]
+    [SerializeField] private float neighbourRadius = 1.5f;
+    [Range(0f, 1f)]
+    [SerializeField] private float avoidanceRadiusMultiplier = 0.5f;
+
+    private float squareMaxSpeed;
+    private float squareNeighbourRadius;
+    public float SquareAvoidanceRadius { get; set; }
+
+    void Start()
+    {
+        this.squareMaxSpeed = this.maxSpeed * this.maxSpeed;
+        this.squareNeighbourRadius = this.neighbourRadius * this.neighbourRadius;
+        SquareAvoidanceRadius = this.squareNeighbourRadius * this.avoidanceRadiusMultiplier * this.avoidanceRadiusMultiplier;
```

Comment:
you could use mathf.pow if u want

Reply:
Made it this way to avoid using math libraries

Reply:
Sure! I was wondering if mathf.pow would be better optimized but oh well